fixtures/cloudflare-dev-module-resolution/index.test.ts

```diff
@@ -0,0 +1,50 @@
+import { createServer } from 'vite';
+import { beforeAll, describe, expect, test } from 'vitest';
+
+import config from './vite.config';
+
+describe('Cloudflare dev module resolution', () => {
+  let jsonOutput: Record<string, unknown> = {};
+
+  beforeAll(async () => {
+    jsonOutput = (await fetchJsonFromViteDevServer()) ?? {};
+  });
+
+  test('can successfully import "React"', () => {
+    expect(jsonOutput['typeof React']).toEqual('object');
+    expect(jsonOutput['typeof React.cloneElement']).toEqual('function');
+    expect(jsonOutput['reactVersionsMatch']).toEqual(true);
+  });
+
+  test('can successfully import utilities from "@remix-run/cloudflare"', () => {
+    expect(jsonOutput['type of remix cloudflare json({})']).toEqual('object');
+    expect(jsonOutput['remixRunCloudflareCookieName']).toEqual(
+      'my-remix-run-cloudflare-cookie',
+    );
+  });
+});
```

Comment:
I think it's best to avoid explicit types when we can't guarantee that they are correct.

```suggestion
describe('Cloudflare dev module resolution', () => {
  let output: unknown
  
  beforeAll(async () => {
    output = await fetchOutputFromViteDevServer();
  });

  test('can successfully import "React"', () => {
    expect(output?.['typeof React']).toEqual('object');
    expect(output?.['typeof React.cloneElement']).toEqual('function');
    expect(output?.['reactVersionsMatch']).toEqual(true);
  });

  test('can successfully import utilities from "@remix-run/cloudflare"', () => {
    expect(output?.['type of remix cloudflare json({})']).toEqual('object');
    expect(output?.['remixRunCloudflareCookieName']).toEqual(
      'my-remix-run-cloudflare-cookie',
    );
  });
});
```

Reply:
Sounds good to me 🙂👍 

Honestly I didn't know you could do `?.[...]` on `unknown`s, that's neat 😄

Reply:
Did you mean to make the suggested changes here?

Reply:
damn, sorry I thought I did 😓 (gh suggestions were really unhelpful/confusing in this PR (because of overlaps I think) 😓)

I just applied the change! thanks for noticing this! 👍

Reply:
I see the test failed because this reverted the change from 'type of remix' to 'typeof remix'. I'll fix it and then merge if the test passes.

Reply:
Ah, you already did it.